controllers/mirrorpeersecret_controller.go

```diff
@@ -0,0 +1,451 @@
+		matchingSourceSecret := findAMatchingSecretWithPeerRef(eachPeerRef, sourceSecrets)
+		// if no match found (ie; no source|BLUE secret found); just continue
+		if matchingSourceSecret == nil {
+			continue
+		}
+		err = r.updateDestinationSecretsFromSource(ctx, matchingSourceSecret, mirrorPeerObj)
+		if err != nil {
+			logger.Error(err, "Error while updating Destination secrets", "source-secret", *matchingSourceSecret)
+			anyErr = err
+		}
+	}
+	return anyErr
+}
+
+func (r *MirrorPeerSecretReconciler) fetchAllInternalSecretsInNamespace(ctx context.Context, namespace string, secretLabelType secretutils.SecretLabelType) ([]corev1.Secret, error) {
+	var sourceSecretList corev1.SecretList
+	var inNS client.InNamespace = client.InNamespace(namespace)
+	var labelOption client.ListOption
+
+	labelOption = client.MatchingLabels(map[string]string{
+		secretutils.SecretLabelTypeKey: string(secretLabelType)})
+	// if the 'secretLabelType' is asking to ignore, then
+	// don't check the label value
+	// just check whether the secret has the internal label
+	if secretLabelType == secretutils.IgnoreLabel {
+		labelOption = client.HasLabels([]string{secretutils.SecretLabelTypeKey})
+	}
+	// find all the secrets with the provided internal label
+	err := r.List(ctx, &sourceSecretList, inNS, labelOption)
+	return sourceSecretList.Items, err
+}
+
+func (r *MirrorPeerSecretReconciler) FetchAllSourceSecretsInNamespace(ctx context.Context, namespace string) ([]corev1.Secret, error) {
+	return r.fetchAllInternalSecretsInNamespace(ctx, namespace, secretutils.SourceLabel)
+}
+
+func (r *MirrorPeerSecretReconciler) FetchAllDestinationSecretsInNamespace(ctx context.Context, namespace string) ([]corev1.Secret, error) {
+	return r.fetchAllInternalSecretsInNamespace(ctx, namespace, secretutils.DestinationLabel)
+}
+
+func (r *MirrorPeerSecretReconciler) mirrorPeerSecretReconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx, "controller", "MirrorPeerSecret")
+	var mirrorPeerSecretObj corev1.Secret
+	err := r.Get(ctx, req.NamespacedName, &mirrorPeerSecretObj)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Info("Secret not found", "req", req)
+			err = r.processDeletedSecrets(ctx, req)
+		}
+		if err != nil {
+			logger.Error(err, "Error in processing deleted secrets", "request", req)
+		}
+		return ctrl.Result{}, err
+	}
+	if secretutils.IsSecretSource(&mirrorPeerSecretObj) {
+		logger.Info("It is a BLUE secret")
+		if err := secretutils.ValidateSourceSecret(&mirrorPeerSecretObj); err != nil {
+			logger.Error(err, "Provided source secret is not valid", "secret", mirrorPeerSecretObj)
+			return ctrl.Result{}, err
+		}
+		err = r.updateDestinationSecretsFromSource(ctx, &mirrorPeerSecretObj)
+		if err != nil {
+			logger.Error(err, "Updating the destination secrets failed", "secret", mirrorPeerSecretObj)
+			return ctrl.Result{}, err
+		}
+	} else if secretutils.IsSecretDestination(&mirrorPeerSecretObj) {
```

Comment:
Made the changes to handle GREEN secret updation.
1. If a BLUE secret is reachable, we update accordingly
2. but if the GREEN is modified in such a way that we cannot reach BLUE, it is deleted. On the next reconcile, it will be re-created from the BLUE if there is a peer mapping.